Documentation/RepoCompose.md

```diff
@@ -0,0 +1,287 @@
+
+### NuGet packages
+
+The description for NuGet artifacts is it two parts:
+
+1. The set of feeds packages are being read from.
+2. The set of packages that are being consumed and their respective versions.
+
+Example:
+
+``` json
+"nuget": {
+    "feeds": [
+        { 
+           "name": "core-clr",
+           "value": "https://dotnet.myget.org/F/dotnet-coreclr/api/v3/index.json" 
+        },
+        {
+            "name": "dotnet-core",
+            "value": "https://dotnet.myget.org/F/dotnet-core/api/v3/index.json"
+        }
+    ],
+    "packages": {
+        "MicroBuild.Core": "0.2.0",
+        "Microsoft.NETCore.Platforms": "1.0.1"
+    }
+}
+```
+
+### File 
+
+Any file which is not a NuGet package should be listed as a file artifact.  These can be downloaded from the web or copied from local places on the hard drive.  Each type of file entry will have a name uniquely identifying the artifact and a kind property specifying the remainder of the properties:
+
+    - uri: a property named `"uri"` will contain an absolute Uri for the artifact.
+    - filesystem: a property named `"location"` will contain an OS specific file path for the artifact.
+
+Example: 
+
+``` json
+"file": {
+    "nuget.exe":  {
+        "kind": "uri",
+        "uri": "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe"
+    }, 
+    "run.exe": { 
+        "kind": "filesystem",
+        "location": "c:\\tools\\run.exe"
+    }
+}
+```
+
+## Samples
+
+### consumes
+
+``` json
+{
+    "os-all": {
+        "dependencies": {
+            "floating": { 
+                 "nuget": {
+                    "feeds": [
+                        { 
+                           "name": "core-clr",
+                           "value": "https://dotnet.myget.org/F/dotnet-coreclr/api/v3/index.json" 
+                        },
+                        {
+                            "name": "dotnet-core",
+                            "value": "https://dotnet.myget.org/F/dotnet-core/api/v3/index.json"
+                        }
+                    ],
+                    "packages" {
+                        "MicroBuild.Core": "0.2.0",
+                        "Microsoft.NETCore.Platforms": "1.0.1"
+                    }
+                },
+                "file": {
+                    "nuget.exe":  {
+                        "uri": "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe"
+                    }
+                }
+            },
+            "toolset": { 
+            },
+            "static": { 
+            }
+        },
+        "prereq": { 
+            "Microsoft Visual Studio": "2015",
+            "CMake" : "1.0"
+        } 
+    }
+}
+```
+
+### produces
+
+``` json
+{
+    "os-all": {
+        "nuget": {
+            "packages" {
+                "MicroBuild.Core": "0.2.0",
+                "Microsoft.NETCore.Platforms": "1.0.1"
+            }
+        },
+        "file": {
+            "nuget.exe": { } 
```

Comment:
This is intentional in the `produces` command.  The documentation for `produces` that the json describes "what" not "where".  

Think of it this way.  The job of Roslyn is to produce the Microsoft.CodeAnalysis nuget package.  Where that package is placed is defined by the arguments to the `publish` command.